docs/06-contributors/999-rfcs/2023-04-16-website-resource.md

```diff
@@ -0,0 +1,111 @@
+---
+title: Website resource
+description: How to implement the website resource
+---
+
+# Website resource
+
+- **Author(s):**: @tsuf239
+- **Submission Date**: 2023-04-16
+- **Stage**: draf
+- **Stage Date**: 2023-04-16
+
+<!---
+Directions for above:
+
+Author(s): Git tag for PR authors and contributors
+Submission Date: Fill in with date of initial submission, YYYY-MM-DD
+Stage: Fill in with current stage in the RFC lifecycle
+Stage Date: Fill in with date of last stage change
+-->
+
+> Implementing the website resource following the [SDK spec](../../04-reference/wingsdk-spec.md#website).
+
+<!--
+This RFCs have 3 sections: Requirements -> Design -> Implementation. We intentionally start with *Design* since it
+is a great way to introduice the feature to readers.
+-->
+
+## Implementation
+
+### Implementation for aws
+
+Following a few AWS guides on the internet (brought in the end of the section), it seems that the common way to serve a static website is by using a bucket.
+
+After creating a public bucket, we'll upload the folder from the relative path.
+AWS requires a index document name (to serve on the "/" path).
+We'll start by hardcoding it to be "index.html", and can add it later to the website props upon request.
+
+The last thing will be a registration to the chosen domain (if exists).
+
+The website resource contains preflight code only.
+
+#### resources:
+
+[Tutorial: Configuring a static website on Amazon S3](https://docs.aws.amazon.com/AmazonS3/latest/userguide/HostingWebsiteOnS3Setup.html)
+[Use your domain for a static website in an Amazon S3 bucket](https://docs.aws.amazon.com/Route53/latest/DeveloperGuide/getting-started-s3.html)
+
+### Implementation for the simulator
+
+The website will be implemented by a static express server:
+
+```js
+const app = Express();
+
+app.use(express.static("<resolved path to the website files folder>"));
+
+app.listen(
+  "<port, ideally will be specified by the user, otherwise generated>"
+);
+```
+
+Since the website is served on the local machine, we cannot specify a domain while deploying to the simulator.
```

Comment:
```suggestion
Since the website is served on the local machine, we cannot specify a domain, nor can we add a CDN while deploying to the simulator.
```